src/dataset/composer.py

```diff
@@ -34,3 +34,45 @@ def __call__(self, image: np.ndarray, annotations: Dict) -> Any:
             image, annotations = func(image, annotations, **self.kwargs)
 
         return image, annotations
+
+
+def join_patches(patches: List[np.ndarray], filenames: List[str]) -> np.ndarray:
+    if len(patches) == 0:
+        raise ValueError("Input list of patches is empty.")
+
+    patches_positions = [x_y_from_filename(filename) for filename in filenames]
+    result_shape = calculate_result_shape_from_patches(patches, patches_positions)
+    result_image = np.zeros(result_shape, dtype=np.uint8)
+
+    for (y_at_result_image, x_at_result_image), patch in zip(patches_positions, patches):
+        height, width = patch.shape
+
+        result_image[
+            y_at_result_image:y_at_result_image + height,
+            x_at_result_image:x_at_result_image + width
+        ] = np.maximum(
+            result_image[y_at_result_image:y_at_result_image + height, x_at_result_image:x_at_result_image + width],
+            patch)
```

Comment:
Alternative way (considering input arrays are always uint8). Don't need to change yours.

```python
 result_image[y_at_result_image:y_at_result_image + height, x_at_result_image:x_at_result_image + width] &= patch
```

Reply:
Sorry. You're right.

Reply:
I had to force, on my original test for this function, that the input arrays are of the correct type with `.astype(np.uint8)`. Example:

```python
image_4x3 = np.array([
                        [1, 1, 1, 1],
                        [1, 1, 1, 1],
                        [1, 1, 1, 1]]).astype(np.uint8)
```

Is this a problem? Should I check on the `join_patches` function for the type being passed? Otherwise this bitwise operation will fail. I could also cast, I guess. WDYT?

Reply:
Casting would be something like this, I think: `patches = [patch.astype(np.uint8) for patch in patches]`

Reply:
It makes sense for me casting them inside `join_patches()` :+1: